django/contrib/gis/db/backends/postgis/schema.py

```diff
@@ -32,15 +33,17 @@ def _create_index_sql(self, model, *, fields=None, **kwargs):
         elif field.dim > 2 and not field.geography:
             # Use "nd" ops which are fast on multidimensional cases
             field_column = "%s %s" % (field_column, self.geom_index_ops_nd)
-        if kwargs.get('name') is None:
-            index_name = '%s_%s_id' % (model._meta.db_table, field.column)
-        else:
-            index_name = kwargs['name']
+
+        name = kwargs.get("name")
+        if not name:
+            name = self._create_index_name(
+                table, [field.column], kwargs.get("suffix", "")
+            )
 
         return Statement(
             self.sql_create_index,
-            name=self.quote_name(index_name),
-            table=self.quote_name(model._meta.db_table),
+            name=self.quote_name(name),
+            table=self.quote_name(table),
```

Comment:
I think we'll want to use `Table` and `IndexName`

https://github.com/django/django/blob/65a9d0013d202447dd76a9cb3c939aa5c9d23da3/django/db/backends/base/schema.py#L1033-L1042

Otherwise column renames and deletion in the same migration will break.

Reply:
Actually I think it might be possible to reuse most of `super()._create_index_sql` by using `expressions=[RawSQL(...)]` instead of `columns` to avoid heavy duplication between both methods.

Reply:
I don't think that a separate ticket is necessary, using `super()._create_index_sql()` will fix described issue.

Reply:
Sounds good. I am not entirely sure I understand how to do the expression part.

Reply:
I wasn't sure how to do the expression idea, so I passed in the `columns` kwarg to `_create_index_sql` instead. Thoughts?

Reply:
Nice! Thanks for that - wouldn't have come up with that on my own. Made some tweaks and tests pass locally.